pkg/store/bucket.go

```diff
@@ -343,7 +343,7 @@ func (s *BucketStore) SyncBlocks(ctx context.Context) error {
 		bdir := path.Join(s.dir, id.String())
 		meta, err := loadMeta(ctx, s.logger, s.bucket, bdir, id)
 		if err != nil {
-			return errors.Wrap(err, "load meta")
+			return errors.Wrap(os.RemoveAll(bdir), "load meta")
```

Comment:
Hmm I don’t think this is quite right. This change will suppress the load meta error if the directory cleanup is successful. If we get an error loading metas, we want to return that error, not the error concerning cleanup. Cleanup should be a defer that logs any error.

Reply:
Do we want to return the error? Wouldn't that cause the Iter function to immediately abort the iteration?

Reply:
Aborting the iteration is the current logic

Reply:
Sure.. but it would be aborting an Iter function due to a transient error (block is still being written), when it could continue on syncing the remaining blocks.

This change would bring this more in-line with the functions below it which also log the error and clean up the directory.

Reply:
I’m not necessarily against changing the logic, it simply isn’t what is done today and was not mentioned in your PR. Nevertheless, this change still suppresses the cause of the error without logging it. It also mixes up the causes of the errors. It would wrap a directory deletion error with `load meta` as though the deletion were the cause of failing to load the meta.

Reply:
I understand what you're saying. I've updated it to be more in-line with the functions below it. Log the error, and return the `os.RemoveAll` error.